### PR TITLE
Fix rocket stages and preflight checks bugs

### PR DESCRIPTION
**Pull request by tornac1234**
- Check for null PreflightChecks
- Prevent a client from sending back the event of completing a preflight check

### CODE REVIEW

**Comment by tornac1234** [comment]
Does it finally fix [Issue#1471](https://github.com/SubnauticaNitrox/Nitrox/issues/1471) ?

File: NitroxClient/Communication/Packets/Processors/RocketPreflightCompleteProcessor.cs [hunk 1]
```diff
@@ -59,5 +62,14 @@ public override void Process(RocketPreflightComplete packet)
             }
 
         }
+
+        private void CompletePreflightCheck(PreflightCheckSwitch preflightCheckSwitch)
+        {
+            packetSender ??= NitroxServiceLocator.LocateService<IPacketSender>();
```
Comment (Jannify) [inline NitroxClient/Communication/Packets/Processors/RocketPreflightCompleteProcessor.cs:68]: You can get assign the packetSender in a constructor in this class.

File: NitroxClient/Communication/Packets/Processors/RocketPreflightCompleteProcessor.cs [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 using NitroxClient.MonoBehaviours;
+using NitroxModel.Core;
 using NitroxModel_Subnautica.Packets;
 using UnityEngine;
 
 namespace NitroxClient.Communication.Packets.Processors
 {
     public class RocketPreflightCompleteProcessor : ClientPacketProcessor<RocketPreflightComplete>
     {
+        private IPacketSender packetSender;
```
Comment (Jannify) [inline NitroxClient/Communication/Packets/Processors/RocketPreflightCompleteProcessor.cs:12]: With the construtor:
```suggestion
        private readonly IPacketSender packetSender;
        
```

**Review by Jannify — CHANGES_REQUESTED** [review]

**Comment by Measurity** [comment]
Is there a reason we don't just instantiate the PreflightChecks so it's never null? 
Example:
```cs
        [ProtoMember(3)]
        public ThreadSafeList<PreflightCheck> PreflightChecks { get; set; } = new();
```

**Review by Measurity — APPROVED** [review]
I don't know much about the rocket stages but the parts I did test worked.

**Review by dartasen — APPROVED** [review]
I'll need to rewrite those processors to use metadata system but for now, it looks good

**Pull request merged**